spec/services/set_participant_categories_spec.rb

```diff
@@ -14,6 +14,13 @@
     let!(:primary_mentor) { create(:participant_profile, :mentor, :ecf_participant_eligibility, :ecf_participant_validation_data, :primary_profile, school_cohort: school_cohort) }
     let!(:secondary_mentor) { create(:participant_profile, :mentor, :ecf_participant_eligibility, :ecf_participant_validation_data, :secondary_profile, school_cohort: school_cohort) }
 
+    before do
+      [primary_mentor, secondary_mentor].each do |profile|
+        profile.ecf_participant_eligibility.determine_status
+        profile.ecf_participant_eligibility.save!
```

Comment:
Happy for this to go in as-is, but I think it would be good to look at the factory to see if this can move to the trait. I wouldn't expect to need to do this, and I think it could lead to other things slipping though

Reply:
Yes agree